src/nspawn/nspawn.c

```diff
@@ -3206,6 +3206,17 @@ static int outer_child(
         return 0;
 }
 
+static int on_sigchld(sd_event_source *s, const struct signalfd_siginfo *si, void *userdata) {
+        pid_t pid;
+
+        pid = PTR_TO_UINT32(userdata);
+        if (pid == si->ssi_pid) {
```

Comment:
This can be reproduced as follows:
```
# systemd-nspawn --directory=/var/lib/toolbox/root-fedora-latest --share-system
Spawning container root-fedora-latest on /var/lib/toolbox/root-fedora-latest.
Press ^] three times within 1s to kill container.
[root@andreyu-coreos ~]# { true & } &
[1] 22201
[root@andreyu-coreos ~]# 
Container root-fedora-latest terminated by signal KILL
```
The first command ("{ true & } &") makes "true" and orpan process. It gets reattached to systemd-nspawn as the parent. 
(why exactly kernel reparents it to nspawn and not to the init(pid=1) as it should normally do is not completely clear to me).